mmdeploy/apis/calibration.py

```diff
@@ -1,107 +1,63 @@
-        >>> from mmdeploy.utils import get_calib_filename, load_config
-        >>> deploy_cfg = 'configs/mmdet/detection/' \
-            'detection_tensorrt-int8_dynamic-320x320-1344x1344.py'
-        >>> deploy_cfg = load_config(deploy_cfg)[0]
-        >>> calib_file = get_calib_filename(deploy_cfg)
-        >>> model_cfg = 'mmdetection/configs/fcos/' \
-            'fcos_r50_caffe_fpn_gn-head_1x_coco.py'
-        >>> model_checkpoint = 'checkpoints/' \
-            'fcos_r50_caffe_fpn_gn-head_1x_coco-821213aa.pth'
-        >>> create_calib_table(calib_file, deploy_cfg, \
-            model_cfg, model_checkpoint, device='cuda:0')
+        device_id = torch.device(device).index
+        if device_id is None:
+            device_id = 0
 
-    Args:
-        calib_file (str): Input calibration file.
-        deploy_cfg (str | mmcv.Config): Deployment config.
-        model_cfg (str | mmcv.Config): The model config.
-        model_checkpoint (str): PyTorch model checkpoint, defaults to `None`.
-        dataset_cfg (str | mmcv.Config): Dataset config, defaults to `None`
-        dataset_type (str): A string specifying dataset type, e.g.: 'test',
-            'val', defaults to 'val'.
-        device (str): Specifying the device to run on, defaults to 'cuda:0'.
-    """
-    if dataset_cfg is None:
-        dataset_cfg = model_cfg
+        # load cfg if necessary
+        deploy_cfg, model_cfg = load_config(deploy_cfg, model_cfg)
 
-    # load cfg if necessary
-    deploy_cfg, model_cfg = load_config(deploy_cfg, model_cfg)
-    device_id = torch.device(device).index
-    if device_id is None:
-        device_id = 0
+        if dataset_cfg is None:
+            dataset_cfg = model_cfg
 
-    if dataset_cfg is None:
-        dataset_cfg = model_cfg
-    # load dataset_cfg if necessary
-    dataset_cfg = load_config(dataset_cfg)[0]
+        # load dataset_cfg if necessary
+        dataset_cfg = load_config(dataset_cfg)[0]
 
-    from mmdeploy.apis.utils import build_task_processor
-    task_processor = build_task_processor(model_cfg, deploy_cfg, device)
+        from mmdeploy.apis.utils import build_task_processor
+        task_processor = build_task_processor(model_cfg, deploy_cfg, device)
 
-    apply_marks = cfg_apply_marks(deploy_cfg)
-    backend = 'default'
-    model = task_processor.init_pytorch_model(model_checkpoint)
-    dataset = task_processor.build_dataset(dataset_cfg, dataset_type)
+        backend = 'default'
+        apply_marks = cfg_apply_marks(deploy_cfg)
 
-    # patch model
-    patched_model = patch_model(model, cfg=deploy_cfg, backend=backend)
+        model = task_processor.init_pytorch_model(model_checkpoint)
+        dataset = task_processor.build_dataset(dataset_cfg, dataset_type)
 
-    with h5py.File(calib_file, mode='w') as file:
-        calib_data_group = file.create_group('calib_data')
+        # patch model
+        patched_model = patch_model(model, cfg=deploy_cfg, backend=backend)
```

Comment:
Is backend here always be `default` ? If so, this parameter could be diminished.

Reply:
Done